examples/leasing/node_release_script.py

```diff
@@ -0,0 +1,187 @@
+    Creates a structure containing all information from the
+    file. We will change and update this data structure and then
+    write it back to the file. In this way, the file will be read
+    and written once.
+    """
+    nodes = {}
+    with open(statusfile, 'r') as status_file:
+        for line in status_file:
+            node_status = line.split()
+            nodes[node_status[0]] = {}
+            nodes[node_status[0]].update({'project': node_status[1],
+                                          'time': node_status[2]})
+    return nodes
+
+
+def release_from_project(
+        hil_client, node, project
+        ):
+    """
+    We need to disconnect all the networks from all of the node's NICs
+    before releasing the node, after that we can detach node from
+    the project
+    """
+
+    node_info = hil_client.node.show(node)
+
+    for nic in node_info['nics']:
+        port = nic['port']
+        switch = nic['switch']
+        if port and switch:
+            try:
+                hil_client.port.port_revert(switch, port)
+                print('Removed all networks from node `%s`' % node)
+            except FailedAPICallException:
+                print('Failed to revert port `%s` on node \
+                        `%s` switch `%s`' % (port, node, switch))
+                raise HILClientFailure()
+
+    time.sleep(5)
+    # tries 2 times to detach the project because there might be a pending
+    # networking action setup (revert port in the previous step).
+    counter = 2
+    while counter:
+
+        try:
+            hil_client.project.detach(project, node)
+            print('Node `%s` removed from project `%s`' % (node, project))
+            break
+        except FailedAPICallException as ex:
+            if ex.message == 'Node has pending network actions':
+                counter -= 1
+                time.sleep(2)
+            else:
+                print('HIL reservation failure: Unable to \
+                        detach node `%s` from project `%s`' % (node, project))
+                raise HILClientFailure(ex.message)
+    if counter == 0:
+        print('HIL reservation failure: Unable to detach node \
+                `%s` from project `%s`' % (node, project))
+        raise HILClientFailure()
+
+
+def update_file(statusfile, nodes):
+    """
+    Writing the updated data structure's info back to the file.
+    """
+
+    with open(statusfile, 'w') as status_file:
+        for node in sorted(nodes):
+            status_file.write(str(node) + ' ' + str(nodes[node]['project']) +
+                              ' ' + str(nodes[node]['time']) + '\n')
+
+
+def release_nodes(
+        statusfile, non_persistent_list,
+        threshold_time, hil_username, hil_password, hil_endpoint,
+        hil_client=None
+        ):
+
+    """Release nodes : After certain amount of time (threshold),
+    we should return the nodes back to the free pool.
+    Tenants who need more time should ask time extension.
+    """
```

Comment:
This isn't the right spot to describe the point of the script; you do that in the readme. Rather, describe, what *this function* does. Include a description of each of the parameters, the return value (if any), and any side effects.